source/plugins/languages/visual/libraries/micropython/visual/definitions_micropython.js

```diff
@@ -0,0 +1,616 @@
+// DO NOT EDIT THIS FILE, IT IS AUTMATICALLY GENERATED
+
+module.exports = function (blockly){
+	var Blockly = blockly.Blockly;
+
+	Blockly.Blocks['micropython_pin'] = {
+		init: function() {
+			this.appendValueInput('GPIO')
+				.setCheck('Number')
+				.appendField('Pin');
+			this.appendDummyInput()
+				.appendField(new Blockly.FieldDropdown([['IN','IN'], ['OUT','OUT'], ['PULL_UP','PULL_UP']]), 'type1');
+			this.appendDummyInput()
+				.appendField(new Blockly.FieldDropdown([['none','NONE'], ['IN','IN'], ['OUT','OUT'], ['PULL_UP','PULL_UP']]), 'type2');
+			this.setInputsInline(true);
+			this.setOutput(true, null);
+			this.setColour(255);
+			this.setTooltip('');
+			this.setHelpUrl('');
+		}
+	};
```

Comment:
Create separate blocks for LED, PWM and ADC. They all use pins, but for LED the pin is only OUT, for PWM the same, for ADC it is only IN. Buttons require PULL_UP or PULL_DOWN, the others don't.